Knoema.Client/Client.cs

```diff
@@ -117,12 +117,18 @@ private Task<T> ApiPost<T>(string path, object obj)
 
 		public Task<IEnumerable<Dataset>> ListDatasets(string source = null, string topic = null, string region = null)
 		{
-			var query = string.Empty
-				.AddUrlParam("source", source)
-				.AddUrlParam("topic", topic)
-				.AddUrlParam("region", region);
 
-			return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset", query);
+			if (string.IsNullOrEmpty(source ?? topic ?? region))
+				return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset");
+			else
+			{
+				var payload = new Dictionary<string, string>();
```

Comment:
it's just my version of code.

On Fri, Aug 21, 2015 at 5:36 PM, rpanday <notifications@github.com> wrote:

> In Knoema.Client/Client.cs
> <https://github.com/Knoema/knoema-csharp-driver/pull/15#discussion_r37629763>
> :
>
> >
> > -			return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset", query);
> > +			if (string.IsNullOrEmpty(source ?? topic ?? region))
> > +				return ApiGet<IEnumerable<Dataset>>("/api/1.0/meta/dataset");
> > +			else
> > +			{
> > +				var payload = new Dictionary<string, string>();
>
> @AlexPhilippov <https://github.com/AlexPhilippov>
> The GET method is existing. Retained it for backward compatibility as dll
> is public.
> Anonymous types don't accept null value
>
> @polyakova <https://github.com/polyakova>
> object payload, don't tell about how to use filters without example. It is
> also prone to errors IMHO.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Knoema/knoema-csharp-driver/pull/15/files#r37629763>.
>



-- 
Evgeniya Polyakova,
Knoema